### PR TITLE
Don't override cache-control if it has already been set

### PR DESCRIPTION
**Pull request by prevostc**
Resolves https://github.com/fastify/fastify-caching/issues/14

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [?] documentation is changed or added -> not sure how to document this.
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/headers.test.js [hunk 1]
```diff
@@ -139,6 +140,37 @@ test('sets public with max-age and s-maxage header', async (t) => {
...
+    expiresIn: 300,
+    serverExpiresIn: 12345
+  }
+
+  const fastify = Fastify()
+  await fastify.register(fp(async (fastify, opts) => {
+    fastify.addHook('onRequest', async (req, reply) => {
+      reply.header('cache-control', 'do not override')
+    })
+  }))
```
Comment (Eomm) [inline test/headers.test.js:157]:
```suggestion
  fastify.addHook('onRequest', async (req, reply) => {
    reply.header('cache-control', 'do not override')
  })
```

File: test/headers.test.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 const { test } = require('tap')
 const Fastify = require('fastify')
+const fp = require('fastify-plugin')
```
Comment (Eomm) [inline test/headers.test.js:5]:
```suggestion
```

**Review by Eomm — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -75,7 +75,9 @@ function fastifyCaching (instance, options, next) {
     }
 
     instance.addHook('onRequest', (req, res, next) => {
-      res.header('Cache-control', value)
+      if (!res.getHeader('Cache-control')) {
```
Comment (jsumners) [inline index.js:78]: Is it accurate to overwrite the header if it has been set to an empty string?

Comment (prevostc) [reply index.js:78]: Good question, I don't know enough about the fastify ecosystem and common patterns to give an answer.

I would personally give the control back to users with a dedicated option and a good default value tbh (2nd proposition [in the issue](https://github.com/fastify/fastify-caching/issues/14#issuecomment-1467112296))

Comment (Eomm) [reply index.js:78]: Let's use `res.hasHeader` instead

Comment (prevostc) [reply index.js]: deal 👍

File: index.js [hunk 4]
```diff
@@ -75,7 +75,9 @@ function fastifyCaching (instance, options, next) {
     }
 
     instance.addHook('onRequest', (req, res, next) => {
-      res.header('Cache-control', value)
+      if (!res.hasHeader('Cache-control')) {
```
Comment (jsumners) [inline index.js:78]:
```suggestion
      if (res.hasHeader('Cache-control') === false) {
```

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**